pyseir/rt/patches.py

```diff
@@ -39,8 +39,9 @@ def f(x):
         weighted_data = dict(
             Rt_MAP_composite=np.average(x["Rt_MAP_composite"], weights=x["population"]),
             Rt_ci95_composite=np.average(x["Rt_ci95_composite"], weights=x["population"]),
+            date=x.name,
         )
 
-        return pd.Series(weighted_data, index=["Rt_MAP_composite", "Rt_ci95_composite"])
+        return pd.Series(weighted_data)
 
-    return combined_df.groupby(combined_df.index).apply(f)
+    return combined_df.groupby(CommonFields.DATE).apply(f)
```

Comment:
I've always found this type of solution of applying a function to a pandas groubpy object hard to reason about. Especially because what is passed to the mapped function changes based on using aggregate vs. apply. So if you have a way to do this that's more easy to reason and read without starting up an interpreter, please let me know.

Reply:
There is https://stackoverflow.com/a/31521177/341400
```
wm = lambda x: np.average(x, weights=df.loc[x.index, "adjusted_lots"])
df.groupby(["contract", "month", "year", "buys"]).agg(adjusted_lots=("adjusted_lots", "sum"),  
                                                      price_weighted_mean=("price", wm))
```
but I find using df.loc in the lambda less clear than groupby apply reading the group.